analysis.md

```diff
@@ -0,0 +1,79 @@
+# Malware Analysis
+
+I'm trying to learn more about PowerShell so I can apply the knowledge towards feature engineering for a machine learning pipeline. I asked for interesting samples and my manager sent me a malicious Word document which had some PowerShell. The file hash of the malicious document is `340795d1f2c2bdab1f2382188a7b5c838e0a79d3f059d2db9eb274b0205f6981`.
+
+In this article, I'll go through the results of my analysis and talk about the methods I used and thought processes at each step.
+
+## Extracting the Source Code
+I used [oletools](https://github.com/decalage2/oletools) to extract the VBA source code from the macro. After installing oletools, simply run the command `olevba <filepath>` to reveal the macro.
+
+## Layer 1 - VB and Embedded PowerShell
+Right off the bat there are several things the attacker does to make the script difficult to unravel. The author embeds a lot of newline spacing, widley spreading apart contents to make important code harder to spot. After revealing the macro with oletools, you'll probably have to do a _lot_ of scrolling to read the whole script. Even worse, if the macro is revealed through a terminal with a limited scrollback buffer, part of the script would be hidden.
+
+There is a function `ParsingA()` that appears to download content from the Internet, but it never seems to be used. This is just an attempt to redirect attention to something unimportant and distract from the actual payload.
+
+You'll also notice that the author uses gratuitous string concatenation to break simple string signatures. This helps the attacker evade malware analysis tools that may look for strings like `"winmgmts:\\.\root\cimv2"` and be thrown off by something like 
+```
+"w" & "" & "in" & "" & "mgm" & "" & "ts" & "" & ":" & "" & "\\" & "." & "\r" & "" & "oot\c" & "" & "imv" & "" & "2"
+```
+
+Also notice the large Base64 blob that `lStr` contains. This blob is decoded and decompressed it for execution. Let's dive further.
```

Comment:
Since the reader doesn't have the Base64 blob in front of them "Also notice" probably isn't the best choice. Assume the reader isn't following along, but give enough detail that they _could_. So, in this case, try something like "there's also a huge string wrapped in lStr which ...".

When you say "The blob is decoded" it's unclear if you mean "during normal execution of the malicious code" or "I decoded the blob".

Say you saw a huge string, it looked like base64, so you decoded it with a script, and then there was another _binary_ blob hiding under that, which you figured out was gzip by asking someone, learning about magic bytes, and learning about the file command. It's ok to admit you learned things for this type of audience. (You already say some of this stuff, but think about grouping it together)

Reply:
Consider adding the  first few lines of the `lStr` blob so it's more obvious what you're talking about.